src/actions/pubSubHandlers.js

```diff
@@ -173,22 +164,51 @@ export function handleLeftChannel({user: userId, channel: channelId}) {
     const users = usersSelector(getState())
     const currentUser = userSelector(getState())
     const isCurrentUser = currentUser.id === userId
-    const user = isCurrentUser ? currentUser : find(users, ({id}) => id === userId)
-
-    dispatch({
-      type: types.REMOVE_USER_FROM_CHANNEL,
-      payload: {
-        channelId,
-        user,
-        isCurrentUser
-      }
-    })
+    const user = isCurrentUser ? currentUser : find(users, ({partner}) => partner.id === userId)
+    if (user) {
+      dispatch({
+        type: types.REMOVE_USER_FROM_CHANNEL,
+        payload: {
+          channelId,
+          user,
+          isCurrentUser
+        }
+      })
+    }
 
     const rooms = joinedRoomsSelector(getState())
     if (!rooms.length) dispatch(goTo('/chat'))
   }
 }
 
+const newNotification = (notification, channel) => (dispatch, getState) => {
+  const users = usersSelector(getState())
+  dispatch({
+    type: types.HANDLE_NOTIFICATION,
+    payload: {
+      ...notification,
+      channel,
+      inviter: find(users, {id: notification.inviterId})
+    }
+  })
+}
+
+export function handleNotification(notification) {
+  return (dispatch, getState) => {
+    const state = getState()
+    const channels = channelsSelector(state)
+    const channel = find(channels, {id: notification.channelId})
+    if (channel) {
+      dispatch(newNotification(notification, channel))
+      return
+    }
+    dispatch(addNewUser(notification.author.id))
```

Comment:
dispatch should not return anything…

Reply:
I doubt that the return value from addNewUser action creator actually forwards the return to the dispatch function, but even if it does, to me dispatch() is like an even emitter, you should not expect any return values there.